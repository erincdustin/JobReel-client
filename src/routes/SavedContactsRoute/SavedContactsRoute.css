.SavedContactsRoute {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
}

.savedContBack {
  position: relative;
  grid-column-start: 1;
  grid-column-end: 9;
}

.savedContBack > img {
  height: 250px;
  width: 250px;
  position: absolute;
  top: -170px;
  left: 5%;
  z-index: -1;
}