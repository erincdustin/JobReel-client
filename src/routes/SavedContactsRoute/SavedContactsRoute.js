import React from 'react';
import SideNav from '../../components/SideNav/SideNav';
import AddContactForm from '../../components/AddContactForm/AddContactForm';
import jobReelApiService from '../../services/jobreel-api-service';
import JobReelContext from '../../context/JobReelContext';
import './SavedContactsRoute.css';
import ContactSummary from '../../components/ContactSummary/ContactSummary'

class SavedContactsRoute extends React.Component {
  
  static contextType = JobReelContext

  componentDidMount = () => {
    jobReelApiService.getSavedContacts()
      .then(res => {
        this.context.setContacts(res);
      }) 
  }

  render() {
    let display;
    (this.context.manualContactAdd === false) ? display = <ContactSummary /> : display = <AddContactForm /> 
    return (
      <div className="SavedContactsRoute">
        <div className='title'>
          <h2>Contacts</h2>
        </div>
        <SideNav />
        {display}
      </div>
    )
  }
}

export default SavedContactsRoute;