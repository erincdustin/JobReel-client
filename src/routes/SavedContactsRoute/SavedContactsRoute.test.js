import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import SavedContactsRoute from './SavedContactsRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('SavedContactsRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SavedContactsRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('savedContactsRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
        <savedContactsRoute />
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})