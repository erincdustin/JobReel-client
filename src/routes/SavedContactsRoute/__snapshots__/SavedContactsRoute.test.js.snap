// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`savedContactsRoute snapshot test renders the UI as expected 1`] = `<savedContactsRoute />`;
